src/lib/import_export/csv_meta.cpp

```diff
@@ -0,0 +1,103 @@
+         "CSV meta file, chunk_size must be an integer.");
+  Assert(static_cast<int64_t>(chunk_size) >= 0, "CSV meta file, chunk_size must not be negative.");
+  meta.chunk_size = chunk_size;
+
+  // Apply only parts of the ParseConfig that are provided, use default values otherwise
+  ParseConfig config{};
+  if (json.find("config") != json.end()) {
+    Assert(json.at("config").is_object(), "CSV meta file,\"Config\" field has to be a json object.");
+    nlohmann::json config_json = json.at("config");
+    assign_if_exists(config.delimiter, config_json, "delimiter");
+    assign_if_exists(config.separator, config_json, "separator");
+    assign_if_exists(config.quote, config_json, "quote");
+    assign_if_exists(config.escape, config_json, "escape");
+    assign_if_exists(config.delimiter_escape, config_json, "delimiter_escape");
+    assign_if_exists(config.reject_quoted_nonstrings, config_json, "reject_quoted_nonstrings");
+    assign_if_exists(config.rfc_mode, config_json, "rfc_mode");
+  }
+
+  if (json.find("columns") != json.end()) {
+    Assert(json.at("columns").is_array(), "CSV meta file,\"Columns\" field has to be a json array.");
+    for (const auto& column : json.at("columns")) {
+      ColumnMeta column_meta;
+      column_meta.name = column.at("name");
+      column_meta.type = column.at("type");
+      assign_if_exists(column_meta.nullable, column, "nullable");
+      meta.columns.push_back(column_meta);
+    }
+  }
+
+  meta.config = config;
+}
+
+void to_json(nlohmann::json& json, const CsvMeta& meta) {
+  nlohmann::json config = nlohmann::json{{"delimiter", std::string(1, meta.config.delimiter)},
+                                         {"separator", std::string(1, meta.config.separator)},
+                                         {"quote", std::string(1, meta.config.quote)},
+                                         {"escape", std::string(1, meta.config.escape)},
+                                         {"delimiter_escape", std::string(1, meta.config.delimiter_escape)},
+                                         {"reject_quoted_nonstrings", meta.config.reject_quoted_nonstrings},
+                                         {"rfc_mode", meta.config.rfc_mode}};
+
+  nlohmann::json columns = "[]"_json;
+  for (const auto& column_meta : meta.columns) {
+    columns.emplace_back(
+        nlohmann::json{{"name", column_meta.name}, {"type", column_meta.type}, {"nullable", column_meta.nullable}});
+  }
+
+  json = nlohmann::json{{"chunk_size", meta.chunk_size}, {"config", config}, {"columns", columns}};
+}
+
+bool operator==(const ColumnMeta& lhs, const ColumnMeta& rhs) {
+  bool equal = lhs.name == rhs.name && lhs.type == rhs.type && lhs.nullable == rhs.nullable;
+  return equal;
+}
+
+bool operator==(const ParseConfig& lhs, const ParseConfig& rhs) {
+  bool equal = lhs.delimiter == rhs.delimiter && lhs.separator == rhs.separator && lhs.quote == rhs.quote &&
```

Comment:
This might be easier to read with `std::tie`.

Reply:
Yes, thanks!